third_party/move/testing-infra/transactional-test-runner/src/vm_test_harness.rs

```diff
@@ -463,19 +481,59 @@ pub fn run_test(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
     run_test_with_config(TestRunConfig::CompilerV1, path)
 }
 
+fn precompiled_v1_stdlib_if_needed(
+    config: &TestRunConfig,
+) -> Option<&'static FullyCompiledProgram> {
+    match config {
+        TestRunConfig::CompilerV1 { .. } => PRECOMPILED_MOVE_STDLIB.as_ref(),
+        TestRunConfig::ComparisonV1V2 { .. } => PRECOMPILED_MOVE_STDLIB.as_ref(),
+        TestRunConfig::CompilerV2 { .. } => None,
+    }
+}
+
+fn precompiled_v2_stdlib_if_needed(
+    config: &TestRunConfig,
+) -> Option<&'static PrecompiledFilesModules> {
+    match config {
+        TestRunConfig::CompilerV1 { .. } => None,
+        TestRunConfig::ComparisonV1V2 { .. } => Some(&*PRECOMPILED_MOVE_STDLIB_V2),
+        TestRunConfig::CompilerV2 { .. } => Some(&*PRECOMPILED_MOVE_STDLIB_V2),
+    }
+}
+
 pub fn run_test_with_config(
-    config: TestRunConfig,
+    mut config: TestRunConfig,
     path: &Path,
 ) -> Result<(), Box<dyn std::error::Error>> {
-    run_test_impl::<SimpleVMTestAdapter>(config, path, Some(&*PRECOMPILED_MOVE_STDLIB), &None)
+    if read_bool_env_var(MOVE_COMPILER_V2) && !matches!(config, TestRunConfig::CompilerV2 { .. }) {
```

Comment:
Do we have to use the environment variable here?

Reply:
This is the entry point from transactional-tests under various dirs (move-bytecode-verifier, move-ir-compiler, move-vm), so if we want to use V2 in those tests we need to have the check somewhere, and we need it before the precompiled V1 libs are compiled.